rr/ai/ai_manager.py

```diff
@@ -158,15 +167,33 @@ def process_image(self, image, model, disp_width, disp_height):
             If couldn't get the image
         """
 
-        image = image.get_data()
+        gst_media = image.get_media()
 
-        image_preprocessed = self.preprocess_detection(image)
+        img = ImageHandler.buffer_to_np_array(
+            image.get_data(), image.get_width(), image.get_height())
 
-        inference_results = self.run_inference(image_preprocessed)
+        image_preprocessed = self.preprocess_detection(img)
 
-        self.on_new_prediction_cb(image_preprocessed, inference_results)
+        inference_results = self.run_inference(image_preprocessed)
 
         image_postprocessed = self.postprocess_detection(
-            image, inference_results)
-
-        self.on_new_postprocess_cb(image_preprocessed)
+            img, inference_results)
+
+        # Create GstBuffer from postprocess image
+        h, w, c = image_postprocessed.shape
+        size = h * w * c
+        buffer = GstUtils.buffer_new_wrapped_full(
+            image_postprocessed.tobytes(), size)
+
+        # Create GstImage
+        _sample = image.get_sample()
```

Comment:
Not needed.

Reply:
It is, actually; because we need to get the caps for the new buffer.